pkg/drivers/libhvee/libhvee_windows.go

```diff
@@ -0,0 +1,309 @@
+	}
+}
+
+func (d *Driver) UpdateConfigRaw(rawConfig []byte) error {
+	var update Driver
+
+	err := json.Unmarshal(rawConfig, &update)
+	if err != nil {
+		return err
+	}
+
+	if update.Memory != d.Memory {
+		log.Debugf("Machine: libhvee -> Updating memory from %d MB to %d MB", d.Memory, update.Memory)
+
+		vm, err := d.getMachine()
+		if err != nil {
+			return err
+		}
+
+		err = vm.UpdateProcessorMemSettings(
+			func(ps *hypervctl.ProcessorSettings) {},
+			func(ms *hypervctl.MemorySettings) {
+				ms.VirtualQuantity = uint64(update.Memory)
+				ms.Limit = uint64(update.Memory)
+			})
+
+		if err != nil {
+			return err
+		}
+	}
+
+	if update.CPU != d.CPU {
+		log.Debugf("Machine: libhvee -> Updating CPU count from %d to %d", d.CPU, update.CPU)
+
+		vm, err := d.getMachine()
+		if err != nil {
+			return err
+		}
+
+		err = vm.UpdateProcessorMemSettings(
+			func(ps *hypervctl.ProcessorSettings) {
+				ps.VirtualQuantity = uint64(update.CPU)
+			},
+			func(ms *hypervctl.MemorySettings) {})
+
+		if err != nil {
+			return err
+		}
+	}
+	if update.DiskCapacity != d.DiskCapacity {
+		if err := d.resizeDisk(int64(update.DiskCapacity)); err != nil {
+			log.Warnf("Machine: libhvee -> Failed to set disk size to %d", update.DiskCapacity)
+			return err
+		}
+	}
+	*d = update
+	return nil
+}
+
+// DriverName returns the name of the driver
+func (d *Driver) DriverName() string {
+	return "libhvee"
+}
+
+func (d *Driver) GetState() (state.State, error) {
+	log.Debugf("Machine: libhvee -> state")
+	vm, err := d.getMachine()
+	if err != nil {
+		return state.Error, err
+	}
+
+	log.Debugf("Machine: libhvee -> state: get")
+	vmState := vm.State()
+	switch vmState {
+	case hypervctl.Enabled:
+		log.Debugf("Machine: libhvee -> state: running")
+		return state.Running, nil
+	case hypervctl.Disabled:
+		log.Debugf("Machine: libhvee -> state: stopped")
+		return state.Stopped, nil
+	}
+
+	log.Debugf("Machine: libhvee -> state: unknown")
+	return state.Error, fmt.Errorf("Unknown state")
+}
+
+// PreCreateCheck checks that the machine creation process can be started safely.
+func (d *Driver) PreCreateCheck() error {
+	// Check that powershell was found
+	if _, err := exec.LookPath("powershell.exe"); err != nil {
+		return ErrPowerShellNotFound
+	}
+
+	// Check that hyperv is installed
+	if err := hypervAvailable(); err != nil {
+		return err
+	}
+
+	// Check that the user is an Administrator
+	isAdmin, err := isAdministrator()
+	if err != nil {
+		return err
+	}
+	if !isAdmin {
+		return ErrNotAdministrator
+	}
+
+	return nil
+}
+
+func (d *Driver) Create() error {
+	log.Debugf("Machine: libhvee -> creating: system settings")
+	systemSettings, err := hypervctl.NewSystemSettingsBuilder().
+		PrepareSystemSettings(d.MachineName, nil).
+		PrepareMemorySettings(
+			func(ms *hypervctl.MemorySettings) {
+				ms.DynamicMemoryEnabled = d.DynamicMemory
+				ms.VirtualQuantity = uint64(d.Memory)
+				ms.Reservation = 1024
+				ms.Limit = uint64(d.Memory)
```

Comment:
I assume that the unit used for memory sizes are the same between crc and libhvee?

Reply:
yes, they are.